packages/region-config-resolver/src/extensions/index.ts

```diff
@@ -0,0 +1,37 @@
+import { AwsRegionExtensionConfiguration } from "@aws-sdk/types";
+import { Provider } from "@smithy/types";
+
+import { getRealRegion } from "../regionConfig/getRealRegion";
+
+/**
+ * @internal
+ */
+export const getAwsRegionExtensionConfiguration = (runtimeConfig: Partial<{ region: string | Provider<string> }>) => {
+  let runtimeConfigRegion: Provider<string> = async () => {
+    const region = runtimeConfig.region!;
```

Comment:
I think `region` needs a runtime check to throw an exception if `region` is `undefined`.